src/DSL.ts

```diff
@@ -0,0 +1,20 @@
+import { queryStrict } from './Query';
+import { taskEither } from 'fp-ts/lib/TaskEither';
+import { refetch } from './Strategy';
+
+export function param<A>() {
```

Comment:
why do we export this (and why the name `param`)?

Reply:
this is for some sort of backward compatibility with "params" in the old DSL:
```ts
compose(product({ a: param<string>(), b: params<number>() }))
```
is more or less the same as
```ts
Query({
  params: { a: t.string, b: t.number }
})
```